var/spack/repos/builtin/packages/variorum/package.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Variorum(CMakePackage):
+    """Variorum is a library providing vendor-neutral interfaces for
+       monitoring and controlling underlying hardware features.
+    """
+    homepage = "https://variorum.readthedocs.io"
+    git = "https://github.com/llnl/variorum.git"
+    url = "https://github.com/LLNL/variorum/archive/v0.1.0.tar.gz"
+
+    maintainers = ["slabasan", "rountree"]
+
+    version("0.1.0", tag="v0.1.0")
+
+    ############
+    # Variants #
+    ############
+    variant("shared", default=True, description="Build Variorum as shared lib")
+
+    # variants for language support
+    variant("python", default=True, description="Build Variorum Python support")
+
+    # variants for dev-tools (docs, etc)
+    variant("docs", default=False, description="Build Variorum's documentation")
+    variant("log", default=False, description="Enable Variorum's logs")
+    variant("debug", default=False, description="Enable Variorum's debug")
+
+    ########################
+    # Package dependencies #
+    ########################
+    depends_on("cmake@2.8:", type="build")
+    depends_on("hwloc@1.11.9")
+
+    #########################
+    # Documentation related #
+    #########################
+    depends_on("py-sphinx", when="+python+docs", type="build")
```

Comment:
What about `+python~docs` or `~python+docs`? This is saying Sphinx is only needed when Python support is requested **and** you want to build the documentation. If you meant **or**, it would look like:
```python
depends_on('py-sphinx', when='+python', type='build')
depends_on('py-sphinx', when='+docs', type='build')
```

Reply:
The Python support is only necessary if the user wants to build the library with the documentation. I think we do want the case where sphinx is only needed if python and build docs are desired.